src/Cli/dotnet/SlnfFileHelper.cs

```diff
@@ -0,0 +1,106 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#nullable disable
+
+using System;
+using System.IO;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli;
+
+/// <summary>
+/// Utilities for working with solution filter (.slnf) files
+/// </summary>
+public static class SlnfFileHelper
+{
+    private class SlnfSolution
+    {
+        [JsonPropertyName("path")]
+        public string Path { get; set; }
+
+        [JsonPropertyName("projects")]
+        public List<string> Projects { get; set; } = new();
+    }
+
+    private class SlnfRoot
+    {
+        [JsonPropertyName("solution")]
+        public SlnfSolution Solution { get; set; } = new();
+    }
+
+    /// <summary>
+    /// Creates a new solution filter file
+    /// </summary>
+    /// <param name="slnfPath">Path to the solution filter file to create</param>
+    /// <param name="parentSolutionPath">Path to the parent solution file</param>
+    /// <param name="projects">List of project paths to include (relative to the parent solution)</param>
+    public static void CreateSolutionFilter(string slnfPath, string parentSolutionPath, IEnumerable<string> projects = null)
+    {
+        var slnfDirectory = Path.GetDirectoryName(Path.GetFullPath(slnfPath)) ?? string.Empty;
+        var parentSolutionFullPath = Path.GetFullPath(parentSolutionPath, slnfDirectory);
+        var relativeSolutionPath = Path.GetRelativePath(slnfDirectory, parentSolutionFullPath);
+
+        // Normalize path separators to backslashes (as per slnf format)
+        relativeSolutionPath = relativeSolutionPath.Replace(Path.DirectorySeparatorChar, '\\');
```

Comment:
There's a lot of path separator normalization going on in this PR. Consider having a common method that provides this logic.